src/psyclone/tests/psyad/transformations/test_assignment_trans.py

```diff
@@ -665,7 +666,14 @@ def test_different_structures(tmpdir):
         "  a%data(n + 1) = a%data(n + 1) + a%data(n)\n"
         "  a%atad(n) = a%atad(n) + a%data(n)\n"
         "  a%data(n) = 0.0\n\n")
-    check_adjoint(tl_fortran, active_variables, ad_fortran, tmpdir)
+
```

Comment:
Thanks for attempting to fix our mess :-) I don't think this test is intended to be specific to the LFRic API. The problem obviously is that it imports `field_type` from `field_mod` and we can't compile that. Please could you revert to just using the generic Compile class and leave the compilation of this xfailing with a TODO that we need to
explicitly define the field_type rather than importing it from a module?

Reply:
OK, I've changed #1530 to correctly describe the problem (field_type should be defined, not imported from LFRic), and reverted LFRicBuild to be Compile (and left it as xfailing). Hope I got it correctly now :)